package/src/utilities/add-vite-plugin.ts

```diff
@@ -1,12 +1,24 @@
 import type { HookParameters } from "astro";
-import type { PluginOption } from "vite";
+import type { Plugin, PluginOption } from "vite";
+import { hasVitePlugin } from "./has-vite-plugin.js";
+
+function incrementPluginName(plugin: Plugin) {
+	let count = 1;
+	plugin.name = `${plugin.name.replace(/-(\d+)$/, (_, c) => {
+		count = parseInt(c) + 1;
+		return "";
+	})}-${count}`;
+}
 
 /**
  * Adds a [vite plugin](https://vitejs.dev/guide/using-plugins) to the
  * Astro config.
  *
  * @param {Params} params
+ * @param {boolean} [params.warnDuplicated=true]
  * @param {import("vite").Plugin} params.plugin
+ * @param {import("astro").HookParameters<"astro:config:setup">["config"]} [params.config]
```

Comment:
I read that wrapping the variable name in brackets `[]` makes it optional, I was trying to make the JSDoc represent the new param type because the `config` and `logger` argument are optional if `warnDuplicated` is `false` but I am not sure how to represent this inside of JSDoc.